client/src/controllers/DrilldownController.ts

```diff
@@ -0,0 +1,58 @@
+import { Controller } from '@hotwired/stimulus';
+
+/**
+ * Instantiates an a11y dialog on the controlled element.
+ * Adds support for hide and show methods and blocking body
+ * scroll when the dialog is open.
+ *
+ * @example
+ * <div
+ *    data-controller="w-dialog"
+ *    data-w-dialog-theme-value="floating"
+ *   >
+ *    <div data-w-dialog-target="body"></div>
+ *  </div>
+ */
+export class DrilldownController extends Controller<HTMLElement> {
+  static targets = ['menu', 'toggle', 'back'];
+
+  declare readonly menuTarget: HTMLElement;
+  declare readonly toggleTargets: HTMLButtonElement[];
+  declare readonly backTargets: HTMLButtonElement[];
+
+  connect() {
+    this.toggleTargets.forEach((toggle) => {
+      toggle.addEventListener('click', this.open.bind(this));
+    });
+
+    this.backTargets.forEach((back) => {
+      back.addEventListener('click', this.close.bind(this));
```

Comment:
using Stimulus actions for this means that you probably will not even need back targets.

```html
                                            <button class="w-drilldown__back" type="button" aria-label="{% trans 'Back' %}" data-action="w-drilldown#close">
                                                {% icon name="arrow-left" %}
                                            </button>
```

For buttons, no need for `click->` it's the default and as noted above, as the `backTarget` is not used for anything but closing the thing, it can just be done with an action.